### ISSUE TITLE
Git application failed because of tls: failed to verify certificate: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by yodatak** [epinio/epinio#2372]
Hi with epinio 1.8.1,
 i cannot git clone my repository because if use a self certificate ( enterprise certificate)
How can i trust it ?

Fetch	Fetch the files at the git repository's branch	
Fail
Get "https://gitlab.XXXX.com/XXXX/XXXXX.git/info/refs?service=git-upload-pack": tls: failed to verify certificate: x509: certificate signed by unknown authority

**Comment by enrichman** [comment]
Hi @yodatak, thanks for reporting this.
At the moment we do not support self signed certificate for git. We need to skip the verification, or provide a way to add to the certificate to the trust store.

---

We need to set this into the `git.CloneOptions` of the `go-git` client.
See https://github.com/epinio/epinio/blob/261777a07255f13cd1f7fc1e0c9d508b99aa3e6f/internal/api/v1/application/importgit.go and https://github.com/go-git/go-git/blob/master/options.go#L75-L78

**Comment by thehejik** [comment]
I've verified that mounting a secret with the gitlab's CA.pem certificate into `epinio-server` pod resolves this issue. So the `dex-tls` approach can be used here.

I did an app deployment from my self-hosted gitlab by using `Git URL` Source type where `https://gitlab.MYDOMAIN.io/epinio/samples.git` was used as URL value.

This can be used as a temporary workaround BUT it is not permanent and won't survive epinio upgrade:
```bash
$ kubectl create secret -n epinio generic gitlab-tls --from-file=tls.crt=gitlab.ca.pem  # tls.crt is used key value; gitlab.ca.pem is the certificate

$ kubectl patch deployment epinio-server -n epinio  -p '{"spec":{"template":{"spec":{"containers":[{"name":"epinio-server","volumeMounts":[{"name":"gitlab-tls","subPath":"tls.crt","mountPath":"/etc/ssl/certs/gitlab-tls.pem"}]}],"volumes":[{"name":"gitlab-tls","secret":{"defaultMode":420,"secretName":"gitlab-tls"}}]}}}}'
```

The patch just mounts the certificate under `/etc/ssl/certs/gitlab-tls.pem` in `epinio-server` pod.

**Comment by enrichman** [comment]
@yodatak fixed with #2488. This feature will be available in the next release (`v1.10.0`)!
